### PR TITLE
Add test for retrieval model with transformer block

### PR DESCRIPTION
**Pull request by oliverholworthy**
Fixes # (issue)

### Goals :soccer:

- Add unit test for retreival model with transformer encoder

### CODE REVIEW

**Comment by oliverholworthy** [comment]
rerun tests

**Comment by oliverholworthy** [comment]
rerun tests

File: merlin/models/tf/outputs/contrastive.py [hunk 1]
```diff
@@ -314,6 +317,8 @@ def sample_negatives(
         return negatives
 
     def embedding_lookup(self, ids: tf.Tensor):
+        if isinstance(ids, tf.RaggedTensor):
+            ids = ids.to_tensor()
```
Comment (marcromeyn) [inline merlin/models/tf/outputs/contrastive.py:321]: The `Embedding` layer from Keras support ragged-tensors, right? What's the reasoning behind always converting to dense?

File: merlin/models/tf/transforms/sequence.py [hunk 2]
```diff
@@ -219,7 +219,7 @@ def call(
         self._check_seq_inputs_targets(inputs)
 
         # Shifts the target column to be the next item of corresponding input column
-        new_target = inputs[self.target_name][:, 1:]
+        new_target = inputs[self.target_name][:, -1:]
```
Comment (sararb) [inline merlin/models/tf/transforms/sequence.py:222]: The tranform `SequencePredictNext` aims to perform a sliding window prediction of size 1 (we referred to it as Causal Language modeling in T4Rec to align with the NLP domain). More specifically, the session-based model will use the hidden representation at position N to predict the target at position N+1. So each input row in the batch will be linked to multiple targets (from position 2 up to last non-padded position)

If you want to predict only the last item of the sequence you should use `SequencePredictLast` indead.

Comment (oliverholworthy) [reply merlin/models/tf/transforms/sequence.py:222]: Thanks for the explanation. I think I got confused here between predict next and predict last.

File: tests/unit/tf/transformers/test_block.py [hunk 3]
```diff
@@ -25,6 +25,59 @@ def test_import():
...
+    output_schema = seq_schema.select_by_name(target)
+
+    query_encoder = mm.Encoder(
+        mm.InputBlockV2(
+            query_schema,
+            embeddings=mm.Embeddings(
+                query_schema.select_by_tag(Tags.CATEGORICAL), sequence_combiner=None
+            ),
+        ),
+        GPT2Block(d_model=48, n_head=4, n_layer=2, pre=mm.ReplaceMaskedEmbeddings()),
```
Comment (gabrielspmoreira) [inline tests/unit/tf/transformers/test_block.py:49]: The `ReplaceMaskedEmbeddings()` is only needed when `SequenceMaskRandom()` is used (Masked Language Modeling. As we are using `SequencePredictNext` here for Causal Language Modeling, there will be no masked items to have embeddings replaced

Comment (oliverholworthy) [reply tests/unit/tf/transformers/test_block.py]: Thanks @gabrielspmoreira that makes sense. I was copying this from another test that used GPT2Block and didn't make the connection with the `SequenceMaskRandom`.

I guess including it in a case like this where there are no masks doesn't affet the model? (in other words does it act as a no-op). Or would it be worth adding something to `ReplaceMaskedEmbeddings` to raise an exception if no masks are found?

**Review by marcromeyn — APPROVED** [review]

**Pull request merged**